pyomo/gdp/util.py

```diff
@@ -134,3 +136,157 @@ def is_child_of(parent, child, knownBlocks=None):
+                    Disjunction
+    """
+    # NOTE: This is indeed a linear search through the Disjunctions on the
+    # model. I am leaving it this way on the assumption that asking XOR
+    # constraints for their Disjunction is not going to be a common
+    # question. If we ever need efficiency then we should store a reverse
+    # map from the XOR constraint to the Disjunction on the transformation
+    # block while we do the transformation. And then this method could query
+    # that map.
+    m = xor_constraint.model()
+    for disjunction in m.component_data_objects(Disjunction):
+        if disjunction._algebraic_constraint:
+            if disjunction._algebraic_constraint() is xor_constraint:
+                return disjunction
+    raise GDP_Error("It appears that %s is not an XOR or OR constraint "
+                    "resulting from transforming a Disjunction."
+                    % xor_constraint.name)
+
+def get_src_disjunct(transBlock):
+    """Return the Disjunct object whose transformed components are on
+    transBlock.
+
+    Parameters
+    ----------
+    transBlock: _BlockData which is in the relaxedDisjuncts IndexedBlock
+                on a transformation block.
+    """
+    try:
+        return transBlock._srcDisjunct()
+    except:
+        raise GDP_Error("Block %s doesn't appear to be a transformation "
+                        "block for a disjunct. No source disjunct found." 
+                        "\n\t(original error: %s)" 
+                        % (transBlock.name, sys.exc_info()[1]))
+
+def get_src_constraint(transformedConstraint):
+    """Return the original Constraint whose transformed counterpart is
+    transformedConstraint
+
+    Parameters
+    ----------
+    transformedConstraint: Constraint, which must be a component on one of 
+    the BlockDatas in the relaxedDisjuncts Block of 
+    a transformation block
+    """
+    transBlock = transformedConstraint.parent_block()
+    # This should be our block, so if it's not, the user messed up and gave
+    # us the wrong thing. If they happen to also have a _constraintMap then
+    # the world is really against us.
+    if not hasattr(transBlock, "_constraintMap"):
+        raise GDP_Error("Constraint %s is not a transformed constraint" 
+                        % transformedConstraint.name)
+    # if something goes wrong here, it's a bug in the mappings.
+    return transBlock._constraintMap['srcConstraints'][transformedConstraint]
+
+def _find_parent_disjunct(constraint):
+    # traverse up until we find the disjunct this constraint lives on
+    parent_disjunct = constraint.parent_block()
+    while not isinstance(parent_disjunct, _DisjunctData):
+        if parent_disjunct is None:
+            raise GDP_Error(
+                "Constraint %s is not on a disjunct and so was not "
+                "transformed" % constraint.name)
+        parent_disjunct = parent_disjunct.parent_block()
+
+    return parent_disjunct
```

Comment:
This is helpful other places, too. I probably have duplicated code in `contrib.gdp_bounds` to accomplish similar functionality.